include/deal.II/lac/vector.templates.h

```diff
@@ -120,23 +120,32 @@ namespace internal
 
   template <typename Functor>
   void vectorized_transform(Functor &functor,
-                            size_type vec_size)
+                            size_type vec_size,
+                            std_cxx11::shared_ptr<parallel::internal::TBBPartitioner> &partitioner)
   {
 #ifdef DEAL_II_WITH_THREADS
     // only go to the parallel function in case there are at least 4 parallel
     // items, otherwise the overhead is too large
     if (vec_size >= 4*internal::Vector::minimum_parallel_grain_size &&
         MultithreadInfo::n_threads() > 1)
       {
+        if (partitioner.get() == NULL)
+          partitioner.reset(new parallel::internal::TBBPartitioner());
```

Comment:
To be honest, I don't know how it happens. I checked that the basic constructors of the method do indeed create a partitioner but I must have forgotten one.

With respect to a race condition: We currently only call this method on vectors that themselves get modified. In case the same vector would be accessed by two threads, it would also race while writing the vector entries. In case we start using this in const functions this will certainly be more problematic (and we will need to make the member variable mutable). I will try to find where it does happen so I can remove the reset. (I observed around 50 test failures when putting an assertion instead of the reset, which means that most of the things were already correct but there is something more.)